cmd/oci/download.go

```diff
@@ -0,0 +1,247 @@
+	// Ensure the repoFlag starts with 'quay.io/'
+	if !strings.HasPrefix(repoFlag, "quay.io/") {
+		return "", "", fmt.Errorf("the repository must start with 'quay.io/'")
+	}
+
+	// Remove 'quay.io/' prefix and split the repo and tag using the ':' character
+	repoFlag = strings.TrimPrefix(repoFlag, "quay.io/")
+	parts := strings.SplitN(repoFlag, ":", 2)
+
+	if len(parts) != 2 {
+		return "", "", fmt.Errorf("tag is missing in the repo flag")
+	}
+
+	return parts[0], parts[1], nil
+}
+
+// parseDuration handles the custom duration format
+func parseDuration(since string) (time.Duration, error) {
+	if len(since) > 1 && since[len(since)-1] == 'd' {
+		days := since[:len(since)-1]
+		hours, err := time.ParseDuration(days + "h")
+		if err != nil {
+			return 0, err
+		}
+		return hours * 24, nil
+	}
+
+	// Parse the duration normally for other time units
+	duration, err := time.ParseDuration(since)
+	if err != nil {
+		return 0, err
+	}
+	return duration, nil
+}
+
+// Init initializes the download command and its flags
+func Init() *cobra.Command {
+	downloadCmd.Flags().StringVar(&opts.repo, "repo", "", "OCI repository and tag to download (e.g., quay.io/test/test:1.0)")
+	downloadCmd.Flags().StringSliceVar(&opts.repos, "repos", nil, "Set of OCI repositories to download from")
+	downloadCmd.Flags().StringVar(&opts.since, "since", "", "Time range to download the latest artifacts (e.g., 4h, 10m, 2d)")
+	downloadCmd.Flags().StringVar(&opts.ociCache, "oci-cache", "", "Directory where OCI artifacts will be cached (default: $HOME/.config/qe-tools/cache)")
+	downloadCmd.Flags().StringVar(&opts.artifactsOutput, "artifacts-output", "", "Mandatory path to store downloaded artifacts")
+	downloadCmd.Flags().BoolVar(&opts.noCache, "no-cache", true, "If true, removes the OCI cache after downloading artifacts")
+
+	// Custom Help function for the download command
+	downloadCmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
+		fmt.Println(`
+Download artifacts from OCI storage using either a single repository or multiple repositories.
+
+Usage:
+  qe-tools download [flags]
+
+Available Flags:
+  --repo             Single OCI repository to download from (e.g., quay.io/test/test:1.0)
+  --repos            Multiple OCI repositories to download from (use with --since)
+  --since            Time range to download the latest artifacts (e.g., 4h, 10m, 2d)
+  --oci-cache        Directory where OCI artifacts will be cached (default: $HOME/.config/qe-tools/cache)
+  --artifacts-output Mandatory path to store downloaded artifacts
+  --no-cache         If true, removes the OCI cache after downloading artifacts
+
+Examples:
+  Download from a single repository:
+    qe-tools download --repo quay.io/test/test:1.0 --artifacts-output /path/to/output
+
+  Download from multiple repositories within the last 2 days:
+    qe-tools download --repos quay.io/repo1 quay.io/repo2 --since 2d --artifacts-output /path/to/output
+	`)
+	})
```

Comment:
Do we need the custom help function? I think cobra can generate it automatically based on params. The custom help function is one more thing to maintain...